pkg/apis/infinispan/v1/types_util.go

```diff
@@ -155,6 +155,11 @@ func (ispn *Infinispan) ApplyDefaults() {
 	if ispn.Spec.Security.EndpointAuthentication == nil {
 		ispn.Spec.Security.EndpointAuthentication = pointer.BoolPtr(true)
 	}
+	if *ispn.Spec.Security.EndpointAuthentication {
+		ispn.Spec.Security.EndpointSecretName = ispn.GetSecretName()
+	} else if ispn.IsGeneratedSecret() {
```

Comment:
`IsGeneratedSecret` should be changed, it worked on assumption, that for generated secret `endpointSecretName` is empty

Reply:
It also assume that a generated secret has a well-known suffix:
https://github.com/infinispan/infinispan-operator/blob/a4b51250caa91795ae35a50cce95628eaf039a10/pkg/apis/infinispan/v1/types_util.go#L373-L374

so it works on that side, but it also check that authentication is enable that needs a check :)

Reply:
I would remove the IsAuthenticationEnabled check from the IsGeneratedSecret method and explicitly check for it in the caller (1 reference), @dmvolod wdyt?

Reply:
Yes, @rigazilla agree.
In this case we can identify that secret is auto-generated by the name only.

Reply:
@dmvolod you mean assuming that it will never be equal to ""? Actually it should be, but I feel a little thrill

Reply:
Yes, it will never be equal to "" if auth is enabled. Previous logic defines that empty value means auto-generated.